---
title: Integration of omics and clinical data of the hormonal, metabolic, inflammatory,
  and oxidative response in the different macronutrients of the diet
author: "Edmond Geraud"
subtitle: '`r params$subtitulo`'
date: '`r format(Sys.Date(),"%B de %e, %Y")`' 
output:
  prettydoc::html_pretty:
    theme: cayman
    highlight: github
    toc: true
  html_document:
    toc: true
    toc_depth: 2
params:
  
  file_IP: Integromics_IPmarkers.xlsx
  file_metabolome: Integromics_Metabolome.xlsx
  file_general: Integromics_1.xlsx
  file_microbiota : integromics_microbiota.xlsx
  folder.data: ../../../datos
  subtitulo: RESULTADOS DESPUES DE LA REUNION 12/1/22
geometry: margin=2cm
---
```{r}
getwd()
"../../../"
```

```{r setup, include=FALSE}
require(knitr)
# include this code chunk as-is to set options
opts_chunk$set(comment = NA, prompt = TRUE, tidy = T, 
               fig.width = 7, fig.height = 7,echo = TRUE, 
               message = FALSE, warning = TRUE, cache=TRUE, fig_caption = TRUE)
Sys.setlocale("LC_TIME", "C")
```

## Load the libraries

```{r libraries}

list.of.packages <- c("xlsx","kableExtra","dplyr","ggplot2","egg","cowplot","patchwork","gridExtra","UsingR","car","lattice","ggpubr","ggbreak","GGally","reshape2","ggcorrplot","corrplot","rela","ggrepel","factoextra","chemometrics","sparsepca","vegan","ca","ade4","pls","OmicsPLS","psych")
new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[,"Package"])] ## check for packages not installed 
if(length(new.packages)) install.packages(new.packages) ## install packages if necessary
res<-unlist(lapply(list.of.packages, require,character.only = T)) ## load packages needed for the session
# if(any(res==F)){
  list.of.packages[which(res==F)]  ## show those package if you have troubles to install.


list.of.bioc.packages <- c("phyloseq","pcaMethods","ropls","mixOmics","limma","DESeq2")
new.packages.bioc <- list.of.packages[!(list.of.bioc.packages %in% installed.packages()[,"Package"])]
if(length(new.packages.bioc)) BiocManager::install(new.packages.bioc)
res.bioc <- unlist(lapply(list.of.bioc.packages, require,character.only = T)) 
if(any(res.bioc==F)){
  list.of.packages[which(res.bioc==F)]  ## show those package if you have troubles to install.
}
```
# Funciones
```{r functions}


proces_bacteria <- function(datos,tipo){

  Order <- bacteria_phylum.abs$Order
  variables_in_bacteria <- general_data[general_data$Orden %in% Order , ]
  Sample<-variables_in_bacteria$Paciente


  if(tipo=="genera"){

    Order <- Order

    X <- datos[-1,-1]
    X<-apply(X,2,as.numeric)
    colnames(X) <- datos[1,2:ncol(datos)]
    rownames(X) <- Sample
    X.rel <- 100*X/rowSums(X)


  } else if(tipo=="phylum"){

    X <- datos[,-c(1:2)]
    colnames(X) <- colnames(datos)[3:ncol(datos)]
    rownames(X) <- Sample
    X.rel<-100*X/rowSums(X)
  }

  GROUP <-variables_in_bacteria$GROUP
  SEX <- variables_in_bacteria$SEX
  OBESE<- variables_in_bacteria$OBESE

  X<-as.data.frame(X)
  X.rel<-as.data.frame(X.rel)

  X$GROUP<-GROUP
  X$SEX<-SEX
  X$OBESE<-OBESE


  X.rel$GROUP<-GROUP
  X.rel$SEX<-SEX
  X.rel$OBESE<-OBESE

  return(list(abs=X,rel=X.rel))

}


low.count.removal = function(
                        data, # OTU count data frame of size n (sample) x p (OTU)
                        percent=0.01 # cutoff chosen
                        ){
    keep.otu = which(colSums(data)*100/(sum(colSums(data))) > percent)
    data.filter = data[,keep.otu]
    return(list(data.filter = data.filter, keep.otu = keep.otu))
}

low.count.removal.counts = function(
                        data, # OTU count data frame of size n (sample) x p (OTU)
                        abundance=10 # cutoff chosen
                        ){
    keep.otu <- which(colSums(data)>abundance)
    data.filter <-data[,keep.otu]
    return(list(data.filter = data.filter, keep.otu = keep.otu))
}

relative.fn <-function(data){
  
  return(100*data/rowSums(data))
  
}


common.bacteria <- function(datos_i,datos.comunes){
  

  ## retorna los indices con las variables comunes

  interaccion <- interaction(datos.comunes$GROUP,datos.comunes$OBESE)
  datos.comunes$interaccion <- interaccion
  grupo <-levels(interaccion)
  w<-vector(mode="list",length=6)
  names(w)<-levels(interaccion)
  for(i in 1:6){
   w[[names(w)[i]]]<-which(apply(as.matrix(datos_i[
    rownames(datos_i)%in%
      datos.comunes[datos.comunes$interaccion==grupo[i],"Paciente"],]),2,function(x) sum(x)==0))
  }

  w2<-vector(mode="list",length=6)
  names(w2)<-levels(interaccion)
  for(i in 1:6){
   w2[[names(w2)[i]]]<-which(apply(as.matrix(datos_i[
    rownames(datos_i)%in%
      datos.comunes[datos.comunes$interaccion==grupo[i],"Paciente"],]),2,function(x) sum(x)>0))
    }

  w3<-lapply(w2, names)
  common_variables <-Reduce(intersect,x = w3)

  return(common_variables)
}

fn.densidad <- function(datos,fctr){
  column_var=c("interaccion","variable","value")
  columnas <- colnames(datos)
  filas <- rownames(datos)
  X<-data.frame(apply(datos, 2, as.numeric))
  colnames(X)<-columnas
  rownames(X)<-filas
  X$fctr<-fctr
  X.melt <- melt(X)
  colnames(X.melt)<-column_var
  p<-ggplot(data=X.melt,aes(x=value,color=interaccion))+geom_density()
  
  return(p)
}

fn.boxplot<-function(datos,fctr){
  column_var=c("interaccion","variable","value")
  columnas <- colnames(datos)
  filas <- rownames(datos)
  X<-data.frame(apply(datos, 2, as.numeric))
  colnames(X)<-columnas
  rownames(X)<-filas
  X$fctr<-fctr
  X.melt <- melt(X)
  colnames(X.melt)<-column_var
  
  p<-ggplot(X.melt,aes(x=interaccion,y=value,fill=interaccion))+geom_boxplot()+theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
  return(p)
}


graphPCA.methods <- function(datos,datos.common,varianzas){
  
  
  sujetos <-as.data.frame(datos)
  sexo<-datos.common$GROUP
  obesidad<-datos.common$OBESE
  colnames(sujetos)<-c("PC1","PC2")
  p1<-ggplot(sujetos,aes(x=PC1, y=PC2)) +
     geom_hline(yintercept = 0, color = "gray70") +
     geom_vline(xintercept = 0, color = "gray70") +
     geom_point(aes(color = sexo,shape=obesidad,fill=obesidad), alpha = 0.55, size = 3) +
     coord_cartesian(xlim = c(min(sujetos[,1]),max(sujetos[,1]))) +
     scale_color_manual(values=c("black","red","green"))+scale_shape_manual(values=c(17,15))+xlab(paste("PC1",round(varianzas[1],2),"%"))+ylab(paste("PC2",round(varianzas[2],2),"%"))
  return(p1)
}
graphPCA.methods2 <- function(datos,datos.common,varianzas){
  
  
  sujetos <-as.data.frame(datos)
  sexo<-datos.common$GROUP
  obesidad<-datos.common$OBESE
  colnames(sujetos)<-c("PC1","PC2")
  p1<-ggplot(sujetos,aes(x=PC1, y=PC2)) +
     geom_hline(yintercept = 0, color = "gray70") +
     geom_vline(xintercept = 0, color = "gray70") +
     geom_point(aes(color = obesidad,shape=sexo,fill=obesidad), alpha = 0.55, size = 3) +
     coord_cartesian(xlim = c(min(sujetos[,1]),max(sujetos[,1]))) +
     scale_color_manual(values=c("black","red"))+scale_shape_manual(values=c(17,15,14))+xlab(paste("PC1",round(varianzas[1],2),"%"))+ylab(paste("PC2",round(varianzas[2],2),"%"))
  return(p1)
}
plot.methods2 <- function(factores,datos.common){
  
  
  sujetos <-data.frame(Factor1 = factores[,1], Factor2=factores[,2])
  sexo<-datos.common$GROUP
  obesidad<-datos.common$OBESE
  p1<-ggplot(sujetos,aes(x=Factor1, y=Factor2)) +
     geom_hline(yintercept = 0, color = "gray70") +
     geom_vline(xintercept = 0, color = "gray70") +
     geom_point(aes(color = sexo,shape=obesidad,fill=obesidad), alpha = 0.55, size = 3)  +
     scale_color_manual(values=c("black","red","green"))+scale_shape_manual(values=c(17,15))
  return(p1)
}

plot.methods <- function(factores,datos.common){
  
  
  sujetos <-data.frame(Sujetos=1:length(factores),Factor=factores)
  sexo<-datos.common$GROUP
  obesidad<-datos.common$OBESE
  p1<-ggplot(sujetos,aes(x=Sujetos, y=Factor)) +
     geom_hline(yintercept = 0, color = "gray70") +
     geom_vline(xintercept = 0, color = "gray70") +
     geom_point(aes(color = sexo,shape=obesidad,fill=obesidad), alpha = 0.55, size = 3)  +
     scale_color_manual(values=c("black","red","green"))+scale_shape_manual(values=c(17,15))
  return(p1)
}


plot.methods2.ob <- function(factores,datos.common){
  
  
  sujetos <-data.frame(Factor1 = factores[,1], Factor2=factores[,2])
  sexo<-datos.common$GROUP
  obesidad<-datos.common$OBESE
  p1<-ggplot(sujetos,aes(x=Factor1, y=Factor2)) +
     geom_hline(yintercept = 0, color = "gray70") +
     geom_vline(xintercept = 0, color = "gray70") +
     geom_point(aes(color = obesidad,shape=grupo,fill=grupo), alpha = 0.55, size = 3)  +
     scale_color_manual(values=c("black","red"))+scale_shape_manual(values=c(17,15,13))
  return(p1)
}

extract_top_weights <- function(modelo,view,factors,thresh=0.5){
  
  p1<-plot_top_weights(modelo,factors=factors,view=view,scale=T)
top_weights <-as.character(unique(p1$data$feature[p1$data$value>=thresh]))
return(top_weights)
}
fun_restore <- function(modelo,grupo){
  factors <- get_factors(modelo, as.data.frame = T,scale = F,groups = grupo)

weights <- rbind(get_weights(modelo,as.data.frame=T,scale=F,views="metaboloma"),
                 get_weights(modelo,as.data.frame=T,scale=F,views="microbioma"))
                 
factores.recast <-as.data.frame(reshape2::recast(data=factors,sample~factor))
rownames(factores.recast) <- factores.recast$sample
factores.recast <- factores.recast[,-1]
weights.recast <-as.data.frame(reshape2::recast(data=weights,feature~factor))
rownames(weights.recast) <- weights.recast$feature
weights.recast <- weights.recast[,-1]
weights.recast$views <- c(rep("metabooloma",37),rep("microbioma",94))
Y <-(as.matrix(factores.recast) %*% t(weights.recast[,-ncol(weights.recast)]))
return(list(Y=Y,factores=factores.recast,pesos=weights.recast))
}
fn.explore <- function(datos.in){
  datos <- as.data.frame(datos.in)
grupo<-variables_in_bacteria$GROUP
obesidad <- factor(gsub(" ",".",variables_in_bacteria$OBESE))
interaccion<-factor(paste(obesidad,grupo,sep="_"))
datos$grupo <- grupo
datos$obesidad <- obesidad
datos$interaccion <- interaccion
datos$sexo <-variables_in_bacteria$SEX
descarte <--c(dim(datos.in)[2]:(dim(datos.in)[2]+4))
#### gruposde interes
femalesOnly <- datos[datos$sexo=="Female",descarte]
malesOnly <- datos[datos$sexo=="Male",descarte]
females <- datos[datos$grupo=="Female",descarte]
males <- datos[datos$grupo=="Male",descarte]
pcos <- datos[datos$grupo=="PCOS",descarte]
obesos <- datos[datos$obesidad=="Obese",descarte]
No.Obesos <- datos[datos$obesidad=="No.Obese",descarte]
#### interacciones
females.no.obese <- datos[datos$interaccion=="No.Obese_Female",descarte]
females.obese <- datos[datos$interaccion=="Obese_Female",descarte]
males.no.obese <-  datos[datos$interaccion=="No.Obese_Male",descarte]
males.obese <-  datos[datos$interaccion=="Obese_Male",descarte]
pcos.no.obese <- datos[datos$interaccion=="No.Obese_PCOS",descarte]
pcos.obese <- datos[datos$interaccion=="Obese_PCOS",descarte]
out <- list(femalesOnly=femalesOnly,
            malesOnly=malesOnly,
            females=females,
            males=males,
            pcos=pcos,
            obesos=obesos,
            No.Obesos=No.Obesos,
            females.no.obese=females.no.obese,
            males.no.obese=males.no.obese,
            pcos.no.obese=pcos.no.obese,
            females.obese=females.obese,
            pcos.obese=pcos.obese,
           males.obese=males.obese,
           Y=datos[,descarte])
return(out)
}


fn.cor.res2 <- function(m1,m2,top_pesos.names,name1,name2){
## correlacion m1
m1.cor <-  cor(m1[,top_pesos.names],method = "spearman")
## cor test m1
m1.cor.p <-corr.test(m1.cor,method = "spearman")$p
# m1.cor.p[colnames(m1.cor.p) %in% colnames(metaboloma),colnames(m1.cor.p) %in% colnames(metaboloma)]<-1
# m1.cor.p[colnames(m1.cor.p) %in% colnames(genus.abs),colnames(m1.cor.p) %in% colnames(genus.abs)]<-1
m1.cor.p[lower.tri(m1.cor.p)]<-1
diag(m1.cor.p)<-1

m2.cor <-  cor(m2[,top_pesos.names],method = "spearman")
m2.cor.p <-corr.test(m2.cor,method = "spearman")$p
# m2.cor.p[colnames(m2.cor.p) %in% colnames(metaboloma),colnames(m2.cor.p) %in% colnames(metaboloma)]<-1
# m2.cor.p[colnames(m2.cor.p) %in% colnames(genus.abs),colnames(m2.cor.p) %in% colnames(genus.abs)]<-1
m2.cor.p[lower.tri(m2.cor.p)]<-1
diag(m2.cor.p)<-1



corrplot::corrplot(m1.cor,p.mat = m1.cor.p,sig.level = 0.05,insig = "label_sig",title = name1,mar = c(0,0,1,0))
corrplot::corrplot(m2.cor,p.mat = m2.cor.p,sig.level = 0.05,insig = "label_sig",title = name2,mar = c(0,0,1,0))
females.only.cor <- cor(m1[,top_pesos.names])
males.only.cor <- cor(m2[,top_pesos.names])
pls.obj<-pls(t(m1[,top_pesos.names]),t(m2[,top_pesos.names]),ncomp = 1)
cortest <-cor.test(pls.obj$variates$X,pls.obj$variates$Y,method = "spearman")$p.value
corval<-cor(pls.obj$variates$X,pls.obj$variates$Y,method = "spearman")

manteltest <-mantel.test(females.only.cor,males.only.cor,nperm = 10000)$p
wilcoxtest <- wilcox.test(pls.obj$X,pls.obj$Y)$p.value

print(paste0(sum(m1.cor.p<0.05),"m1"))
print(paste0(sum(m2.cor.p<0.05),"m1"))

return(c(cortest=cortest,manteltest=manteltest,wilcoxtest=wilcoxtest,corval = corval))
}

fn.cor.res1 <- function(m1,top_pesos.names,name1){
## correlacion m1
m1.cor <-  cor(m1[,top_pesos.names],method = "spearman")
## cor test m1
m1.cor.p <-corr.test(m1.cor,method = "spearman")$p
# m1.cor.p[colnames(m1.cor.p) %in% colnames(metaboloma),colnames(m1.cor.p) %in% colnames(metaboloma)]<-1
# m1.cor.p[colnames(m1.cor.p) %in% colnames(genus.abs),colnames(m1.cor.p) %in% colnames(genus.abs)]<-1
m1.cor.p[lower.tri(m1.cor.p)]<-1
diag(m1.cor.p)<-1

corrplot::corrplot(m1.cor,p.mat = m1.cor.p,sig.level = 0.05,insig = "label_sig",title = name1,mar = c(0,0,1,0))

}
fn.explore2 <- function(datos.in,variables_integracion){
  datos <- as.data.frame(datos.in)
interaccion<-factor(paste(variables_integracion$obesidad,variables_integracion$grupo,sep="_"))
datos$grupo <- variables_integracion$grupo
datos$obesidad <- variables_integracion$obesidad
datos$interaccion <- interaccion
datos$sexo <-variables_in_bacteria$SEX
descarte <--c(dim(datos.in)[2]:(dim(datos.in)[2]+4))
#### gruposde interes
femalesOnly <- datos[datos$sexo=="Female",descarte]
malesOnly <- datos[datos$sexo=="Male",descarte]
females <- datos[datos$grupo=="Female",descarte]
males <- datos[datos$grupo=="Male",descarte]
pcos <- datos[datos$grupo=="PCOS",descarte]
obesos <- datos[datos$obesidad=="Obese",descarte]
No.Obesos <- datos[datos$obesidad=="No.Obese",descarte]
#### interacciones
females.no.obese <- datos[datos$interaccion=="No.Obese_Female",descarte]
females.obese <- datos[datos$interaccion=="Obese_Female",descarte]
males.no.obese <-  datos[datos$interaccion=="No.Obese_Male",descarte]
males.obese <-  datos[datos$interaccion=="Obese_Male",descarte]
pcos.no.obese <- datos[datos$interaccion=="No.Obese_PCOS",descarte]
pcos.obese <- datos[datos$interaccion=="Obese_PCOS",descarte]
out <- list(femalesOnly=femalesOnly,
            malesOnly=malesOnly,
            females=females,
            males=males,
            pcos=pcos,
            obesos=obesos,
            No.Obesos=No.Obesos,
            females.no.obese=females.no.obese,
            males.no.obese=males.no.obese,
            pcos.no.obese=pcos.no.obese,
            females.obese=females.obese,
            pcos.obese=pcos.obese,
           males.obese=males.obese,
           Y=datos[,descarte])
return(out)
}
fun_restore <- function(modelo,grupo){
  factors <- get_factors(modelo, as.data.frame = T,scale = F,groups = grupo)

weights <- rbind(get_weights(modelo,as.data.frame=T,scale=F,views="metaboloma"),
                 get_weights(modelo,as.data.frame=T,scale=F,views="microbioma"))
                 
factores.recast <-as.data.frame(reshape2::recast(data=factors,sample~factor))
rownames(factores.recast) <- factores.recast$sample
factores.recast <- factores.recast[,-1]
weights.recast <-as.data.frame(reshape2::recast(data=weights,feature~factor))
rownames(weights.recast) <- weights.recast$feature
weights.recast <- weights.recast[,-1]
weights.recast$views <- c(rep("metabooloma",37),rep("microbioma",94))
Y <-(as.matrix(factores.recast) %*% t(weights.recast[,-ncol(weights.recast)]))
return(list(Y=Y,factores=factores.recast,pesos=weights.recast))
}

```


# PREPROCESADO

## Datos clinicos

```{r}
general_data <- read.xlsx(file.path(params$folder.data,params$file_general),sheetIndex =1)
general_data$SEX <- factor(general_data$SEX, levels = c(0, 2), labels = c("Female", "Male"))
general_data$OBESE <- factor(general_data$OBESE, levels = c(0, 1), labels = c("No Obese", "Obese"))
general_data$GROUP <- factor(general_data$GROUP, levels = c(0, 1, 2), labels = c("Female",  "PCOS", "Male"))


```


## Datos IP

```{r}

IP_file<-file.path(params$folder.data,file=params$file_IP) ## file path of the data
IP.tmp<-read.xlsx(IP_file,sheetIndex = 1)
IP<-IP.tmp[,-7] # remove the missin values column
IP$SEX <- factor(IP$SEX,levels=c(0,2),labels = c("Female","Male"))
IP$OBESE <- factor(IP$OBESE,levels=c(0,1),labels = c("No Obese","Obese"))
IP$GROUP <- factor(IP$GROUP,levels=c(0,1,2),labels = c("Female","PCOS","Male"))

```

## Datos Metaboloma

```{r}
metabolome_file<-file.path(params$folder.data,file=params$file_metabolome) ## file path of the data
metabolome.tmp<-read.xlsx(metabolome_file,sheetIndex = 1) # read the data
metabolites_name <- read.xlsx(metabolome_file,sheetIndex = 2)$Metabolitos # read the names of the metabolites
any(is.na(metabolome.tmp)) # there is a row of missing values.
metabolome<-metabolome.tmp[-nrow(metabolome.tmp),] ## remove the missing value row
metabolome$GROUP<-general_data$GROUP ## add GROUP variable 
metabolome$OBESE <- general_data$OBESE ## add OBESE variables.

```


## Datos microbioma

```{r}
bacteria_phylum.abs <- as.data.frame(read.xlsx(file.path(params$folder.data,params$file_microbiota), sheetIndex = 1))
bacteria_genera.abs <- as.data.frame(t(read.xlsx(file.path(params$folder.data,params$file_microbiota), sheetIndex = 3)))
phylum.list <- proces_bacteria(bacteria_phylum.abs,tipo="phylum")
genera.list <- proces_bacteria(bacteria_genera.abs,tipo="genera")

phylum.abs <- phylum.list$abs;phylum.rel<-phylum.list$rel
genera.abs <- genera.list$abs;genera.rel<-genera.list$rel

```

# Preparando los datos para la integracion
```{r}
variables_in_bacteria <- general_data[general_data$Orden %in% bacteria_phylum.abs$Order, ]

```

## Datos Clinicos
**NOTE we are going to include also on both levels of integration body measures**
We exclude ISI because it is measured after the load of glucose, the mean values, to exclude redundancy of the data, ant the postprandial levels. 


```{r}
sujetos <- variables_in_bacteria$Paciente
Clin.tmp <- variables_in_bacteria[,-c(1,2,3,4,5)] # get rid off orden, paciente, sex, group and obese
auc_clin <- colnames(Clin.tmp)[grep("AUC",colnames(Clin.tmp))]
mean_basal_clin<-colnames(Clin.tmp)[grep("_B$",colnames(Clin.tmp))]
sog <- colnames(Clin.tmp)[grep("^SO[GLP]",colnames(Clin.tmp))]
variables_clin <- colnames(Clin.tmp)
testosterona <-colnames(Clin.tmp)[grep("TEST",colnames(Clin.tmp))]
interr<-colnames(Clin.tmp)[grep("interaccion",colnames(Clin.tmp))]
ratio <- colnames(Clin.tmp)[grep("Ratio",colnames(Clin.tmp))]
w<-c(which(variables_clin %in% auc_clin),which(variables_clin %in% mean_basal_clin),which(variables_clin %in% sog),which(variables_clin %in% testosterona),
which(variables_clin %in% interr),which(variables_clin %in% ratio))
Clin <- Clin.tmp[,-w] 
rownames(Clin) <- sujetos
colnames(Clin)
```


## Datos IP
```{r}
IP.tmp <- IP[IP$Paciente %in% variables_in_bacteria$Paciente,]
redundant_variables <- c("Orden","Paciente","SEX","GROUP","OBESE","BMI")
variables_ip <- colnames(IP.tmp)
auc_ip <- variables_ip[grep("AUC",variables_ip)]
mean_ip <- variables_ip[grep("_0$",variables_ip)]
w <- c(which(variables_ip %in% redundant_variables),which(variables_ip %in% auc_ip),which(variables_ip %in% mean_ip))
IP.basal <- IP.tmp[,-w]
rownames(IP.basal) <- sujetos

```

## Metabolome data (37)

```{r}
metabolome.basal.tmp <- metabolome[metabolome$Order %in% variables_in_bacteria$Orden,]
redundant_data_meta <- c("Order","Sample","GROUP","OBESE")
variables_meta <- colnames(metabolome.basal.tmp)
mean_meta <- variables_meta[grep("_[GLP]0$",variables_meta)]
auc_meta <- variables_meta[grep("AU[CG]",variables_meta)]
w <- c(which(variables_meta %in% redundant_data_meta),which(variables_meta %in% mean_meta),which(variables_meta %in% auc_meta))
metabolome.basal_mean <- metabolome.basal.tmp[,-w]
rownames(metabolome.basal_mean) <- sujetos
colnames(metabolome.basal_mean)
```
## Datos microbioma


### Datos genero

```{r}

genera.abs.tmp <- genera.abs
redundant_data_ph <- c("GROUP","SEX","OBESE")
genus.tmp.tmp <- genera.abs.tmp[,-c(ncol(genera.abs.tmp)-2,ncol(genera.abs.tmp)-1,ncol(genera.abs.tmp))]
genus <- genus.tmp.tmp[,colSums(genus.tmp.tmp)>0]
rownames(genus) <- sujetos
## removal of genus unidentified

genus.abs<-genus[,-grep("^ud-",colnames(genus))]


```

# Datos que necesitamos

```{r}

pacientes <- variables_in_bacteria$Paciente
grupo<-variables_in_bacteria$GROUP
obesidad <- factor(gsub(" ",".",variables_in_bacteria$OBESE))
interaccion<-factor(paste(obesidad,grupo,sep="_"))
datos.common <- variables_in_bacteria[,c("GROUP","OBESE","Paciente")]
columnData <- data.frame(interaccion=interaccion)
rownames(columnData) <- pacientes

```


```{r}
met37<-t(metabolome.basal_mean)
met37_norm <- met37/rowSums(met37)
met37_log <- t(apply(met37_norm,1,log))
design <- model.matrix(~obesidad*grupo)
X<-(voom(met37_norm,normalize.method = "quantile",design = design)$E)

library(ALDEx2)
set.seed(123)
conds <- interaccion
x.all <- aldex.clr(t(genus.abs),conds)
```



# Descriptiva

## Metaboloma

```{r}
met37<-t(metabolome.basal_mean)
met37_norm <- met37/rowSums(met37)
met37_log <- t(apply(met37_norm,1,log))
design <- model.matrix(~obesidad*grupo)
X<-(voom(met37_norm,normalize.method = "quantile",design = design)$E)
p1<-fn.densidad((t(met37)),interaccion)
p2<-fn.boxplot(t(met37),interaccion)
pcx <- prcomp(t(X),scale=F,center=T)
varianzas <- 100*(pcx$sdev^2/sum(pcx$sdev^2))
p3<-graphPCA.methods(pcx$x,datos.common,varianzas)+ggtitle("PCA")

U3 <- pcx$x
apply(U3, 2, function(x) which( abs(x - mean(x)) > (6 * sd(x)) ))



pcx.cor <- princomp(x=cor((X),method = "spearman"),cor=T)
varianzas <- 100*(pcx.cor$sdev^2/sum(pcx.cor$sdev^2))
p4<-graphPCA.methods(pcx.cor$scores[,1:2],datos.common,varianzas)+ggtitle("PCA spearman")

plsda.met <- mixOmics::plsda(t(X),obesidad,scale = T,ncomp = 5)
p5<-graphPCA.methods2(plsda.met$variates$X,datos.common,varianzas)+ggtitle("PLS-DA")

p7 <-graphPCA.methods2(pcx$x,datos.common,varianzas)+ggtitle("PCA")

euc <- vegdist(t(X),method = "euclidean")
points<-cmdscale(euc)
p6<-graphPCA.methods(points,datos.common,varianzas)+ggtitle("MDS")
ggarrange(p1,p2,p3,p7,
          common.legend = T)
ggarrange(p3,p4,p5,p6,ncol=2,nrow=2,common.legend = T)
ggarrange(p3,p7)
(res<-adonis2(euc~obesidad*grupo))

```

```{r}
fviz_pca_var(pcx)

```


```{r}
X<-(voom(met37_norm,normalize.method = "quantile",design = design)$E)
```



```{r}
metaboloma.process <- as.data.frame(t(X))

```


```{r}
metaboloma.process$interaccion<-interaccion
```

```{r}
levels(interaccion)
```

```{r}
mywilcox <- function(x){
  
  gi <- levels(grupo_interes$interaccion)
  median1 <- median(x[grupo_interes$interaccion==gi[1]])
  median2 <- median(x[grupo_interes$interaccion==gi[2]])
  pvalue <- wilcox.test(x~grupo_interes$interaccion,exact=F)$p.value

  return(pvalue=pvalue)
}
combinaciones <-t(combn(levels(metaboloma.process$interaccion),2))
matriz <- matrix(NA,nrow=ncol(metaboloma.process)-1,ncol=(nrow(combinaciones)))
for(comb in 1:dim(combinaciones)[1]){
  grupo_interes <- metaboloma.process[metaboloma.process$interaccion==combinaciones[comb,1] | metaboloma.process$interaccion==combinaciones[comb,2],]
grupo_interes$interaccion <- droplevels(grupo_interes$interaccion)

matriz[,comb] <-apply(grupo_interes[,-ncol(grupo_interes)], 2,function(x) mywilcox(x))
  
  
}

colnames(matriz)<-  apply(combinaciones, 1,function(X) paste(X[1],"vs",X[2]))
rownames(matriz) <- colnames(metaboloma.process)[-length(metaboloma.process)]
matriz.adj <- t(apply(matriz, 1, function(x) p.adjust(x,method = "BH")))
# write.csv(matriz.adj,"./comparaciones_BH_metaboloma.csv")
```


```{r}
metaboloma.process$interaccion<-obesidad

mywilcox <- function(x){
  
  gi <- levels(grupo_interes$interaccion)
  median1 <- median(x[grupo_interes$interaccion==gi[1]])
  median2 <- median(x[grupo_interes$interaccion==gi[2]])
  pvalue <- wilcox.test(x~grupo_interes$interaccion,exact=F)$p.value

  return(pvalue=pvalue)
}
combinaciones <-t(combn(levels(metaboloma.process$interaccion),2))
matriz <- matrix(NA,nrow=ncol(metaboloma.process)-1,ncol=(nrow(combinaciones)))
for(comb in 1:dim(combinaciones)[1]){
  grupo_interes <- metaboloma.process[metaboloma.process$interaccion==combinaciones[comb,1] | metaboloma.process$interaccion==combinaciones[comb,2],]
grupo_interes$interaccion <- droplevels(grupo_interes$interaccion)

matriz[,comb] <-apply(grupo_interes[,-ncol(grupo_interes)], 2,function(x) mywilcox(x))
  
  
}

colnames(matriz)<-  apply(combinaciones, 1,function(X) paste(X[1],"vs",X[2]))
rownames(matriz) <- colnames(metaboloma.process)[-length(metaboloma.process)]
matriz.adj <- t(apply(matriz, 1, function(x) p.adjust(x,method = "BH")))
# write.csv(matriz.adj,"./comparaciones_BH_metaboloma.csv")
```

```{r}
metaboloma.process$interaccion<-interaccion

mywilcox <- function(x){
  
  gi <- levels(grupo_interes$interaccion)
  median1 <- median(x[grupo_interes$interaccion==gi[1]])
  median2 <- median(x[grupo_interes$interaccion==gi[2]])
  pvalue <- wilcox.test(x~grupo_interes$interaccion,exact=F)$p.value

  return(pvalue=pvalue)
}
combinaciones <-t(combn(levels(metaboloma.process$interaccion),2))
matriz <- matrix(NA,nrow=ncol(metaboloma.process)-1,ncol=(nrow(combinaciones)))
for(comb in 1:dim(combinaciones)[1]){
  grupo_interes <- metaboloma.process[metaboloma.process$interaccion==combinaciones[comb,1] | metaboloma.process$interaccion==combinaciones[comb,2],]
grupo_interes$interaccion <- droplevels(grupo_interes$interaccion)

matriz[,comb] <-apply(grupo_interes[,-ncol(grupo_interes)], 2,function(x) mywilcox(x))
  
  
}

colnames(matriz)<-  apply(combinaciones, 1,function(X) paste(X[1],"vs",X[2]))
rownames(matriz) <- colnames(metaboloma.process)[-length(metaboloma.process)]
matriz.adj <- (apply(matriz, 1, function(x) p.adjust(x,method = "BH")))
write.csv(matriz.adj,"./comparaciones_BH_metaboloma.csv")
```


```{r}
interes <- (matriz.adj)

resultado <- interes*NA
for(f in 1:nrow(matriz.adj)){
  
  cuestion <- interes[f,]
  for(col in 1:ncol(matriz.adj)){
    if(cuestion[col]<0.05 & all(cuestion[-col]>0.05)){
      resultado[f,col]<- "UNICO"
    }else{
      next
    }
  }
  
}
```



```{r}
interes <- t(interes)
interes2 <- interes[,c("No.Obese_Female vs No.Obese_PCOS",
                       "No.Obese_Female vs Obese_Female",
                       "No.Obese_Female vs Obese_PCOS",
                       "No.Obese_PCOS vs Obese_Female",
                       "No.Obese_PCOS vs Obese_PCOS",
                       "Obese_Female vs Obese_PCOS")]
resultado <- interes2*NA
for(f in 1:nrow(matriz.adj)){
  
  cuestion <- interes2[f,]
  for(col in 1:ncol(interes2)){
    if(cuestion[col]<0.05 & all(cuestion[-col]>0.05)){
      resultado[f,col]<- "UNICO"
    }else{
      next
    }
  }
  
}

```


```{r}
interes2 <- interes[,c("No.Obese_Male vs Obese_Male")]
which(interes2<0.05)
```





```{r}
plotLoadings(plsda.met, contrib = 'max', method = 'median',comp = 1,size.name = 0.7,ndisplay = 10)
plotLoadings(plsda.met, contrib = 'max', method = 'median',comp = 2,size.name = 0.7,ndisplay = 10)


```


```{r}
metaboloma <- as.data.frame(t(X))
```


```{r}
metaboloma.extra <- metaboloma
metaboloma.extra$grupo <- grupo
metaboloma.extra$obesidad <- obesidad
metaboloma.extra$interaccion <- interaccion
```

```{r}
p1 <-  ggboxplot(metaboloma.extra,x="obesidad",y="Valine_mean",color = "grupo")+ylab("Valine")+xlab("")
p2 <-  ggboxplot(metaboloma.extra,x="obesidad",y="Oxoisocaproic_mean",color = "grupo")+ylab("Acido Oxoisocaproico")+xlab("")
p3 <-  ggboxplot(metaboloma.extra,x="obesidad",y="Leu_mean",color = "grupo")+ylab("Leucina")+xlab("")
p4 <-  ggboxplot(metaboloma.extra,x="obesidad",y="Glycerol_mean",color = "grupo")+ylab("Glycerol")+xlab("")
ggarrange(p1,p2,p3,p4,common.legend = T)
```

```{r}
wil <-pairwise.wilcox.test(metaboloma.extra$Glycerol_mean,interaccion)

wil$p.value
```



```{r}
datos <- as.data.frame(metaboloma)
grupo<-variables_in_bacteria$GROUP
obesidad <- factor(gsub(" ",".",variables_in_bacteria$OBESE))
interaccion<-factor(paste(obesidad,grupo,sep="_"))
datos$grupo <- grupo
datos$obesidad <- obesidad
datos$interaccion <- interaccion
descarte <- c(40,39,38)
#### gruposde interes
females <- datos[datos$grupo=="Female",-descarte]
males <- datos[datos$grupo=="Male",-descarte]
pcos <- datos[datos$grupo=="PCOS",-descarte]
obesos <- datos[datos$obesidad=="Obese",-descarte]
No.Obesos <- datos[datos$obesidad=="No.Obese",-descarte]
#### interacciones
females.no.obese <- datos[datos$interaccion=="No.Obese_Female",-descarte]
females.obese <- datos[datos$interaccion=="Obese_Female",-descarte]
males.no.obese <-  datos[datos$interaccion=="No.Obese_Male",-descarte]
males.obese <-  datos[datos$interaccion=="Obese_Male",-descarte]
pcos.no.obese <- datos[datos$interaccion=="No.Obese_PCOS",-descarte]
pcos.obese <- datos[datos$interaccion=="Obese_PCOS",-descarte]
```



```{r}
corrplot::corrplot(corr=cor(females,method = "spearman"),)
```


```{r}
library(ape)
fn.cor.res2(obesos,No.Obesos,top_pesos.names = colnames(females),name1 = "pcos no obese","pcos obesos")
```


```{r}
library(ape)
descarte <- c(40,39,38)
mantel.test(cor(obesos[,-descarte]),cor(No.Obesos[,-descarte]),graph=F)
```


```{r}
g1 <-mantel.test(cor(females[,-descarte]),cor(males[,-descarte]),graph=F)$p
g2 <- mantel.test(cor(females[,-descarte]),cor(pcos[,-descarte]),graph=F)$p
g3 <- mantel.test(cor(pcos[,-descarte]),cor(males[,-descarte]),graph=F)$p
p.adjust(c(g1,g2,g3),method = "bonferroni")
```


```{r}
lista.aux <- list(
  females.no.obese=females.no.obese[,-descarte],
  females.obese=females.obese[,-descarte],
  pcos.no.obese=pcos.no.obese[,-descarte],
  pcos.obese=pcos.obese[,-descarte],
  males.no.obese=males.no.obese[,-descarte],
  males.obese=males.obese[,-descarte]
)

mantelas <- vector("numeric",length=15)
combi <- t(combn(names(lista.aux),2))
for(i in 1:dim(combi)[1]){
  mantelas[i] <- mantel.test(cor(lista.aux[[combi[i,1]]]),cor(lista.aux[[combi[i,2]]]))$p
}

names(mantelas)<-apply(combi, 1, function(x) paste(x,collapse = "_vs_"))
mantelas.adj <-p.adjust(mantelas,method = "bonferroni")
names(mantelas.adj[mantelas.adj<0.05])
```

```{r}

```



```{r}
g1 <-mantel.test(cor(females.no.obese[,-descarte]),cor(females.obese[,-descarte]),graph=F)$p
g2 <- mantel.test(cor(females.no.obese[,-descarte]),cor(pcos.no.obese[,-descarte]),graph=F)$p
g3 <- mantel.test(cor(females.no.obese[,-descarte]),cor(females.obese[,-descarte]),graph=F)$p

g4 <-mantel.test(cor(females.no.obese[,-descarte]),cor(males.obese[,-descarte]),graph=F)$p
g5 <- mantel.test(cor(females.no.obese[,-descarte]),cor(pcos.obese[,-descarte]),graph=F)$p
g6 <- mantel.test(cor(males.no.obese[,-descarte]),cor(pcos.no.obese[,-descarte]),graph=F)$p

g7 <-mantel.test(cor(males.no.obese[,-descarte]),cor(females.no.obese[,-descarte]),graph=F)$p
g8 <- mantel.test(cor(males.no.obese[,-descarte]),cor(males.obese[,-descarte]),graph=F)$p
g9 <- mantel.test(cor(males.no.obese[,-descarte]),cor(pcos.obese[,-descarte]),graph=F)$p


g10 <-mantel.test(cor(pcos.no.obese[,-descarte]),cor(females.obese[,-descarte]),graph=F)$p
g11 <- mantel.test(cor(pcos.no.obese[,-descarte]),cor(pcos.obese[,-descarte]),graph=F)$p
g12 <- mantel.test(cor(pcos.no.obese[,-descarte]),cor(males.obese[,-descarte]),graph=F)$p

g13 <-mantel.test(cor(females.obese[,-descarte]),cor(males.obese[,-descarte]),graph=F)$p
g14 <- mantel.test(cor(females.obese[,-descarte]),cor(pcos.obese[,-descarte]),graph=F)$p
g15 <- mantel.test(cor(males.obese[,-descarte]),cor(pcos.obese[,-descarte]),graph=F)$p
```

```{r}
which(p.adjust(c(g1,g2,g3,g4,g5,g6,g7,g8,g9,g10,g11,g12,g13,g14,g15),method = "bonferroni")>0.05)
```

```{r}
variates <- plsda.met$variates$X
matriz <- matrix(NA,ncol=37,nrow = 5)
matriz.p <-matrix(NA,ncol=37,nrow = 5)
for(i in 1:5){
  for(j in 1:37){
    
    matriz[i,j]<- cor(variates[,i],metaboloma[,j],method = "spearman")
    matriz.p[i,j]<-cor.test(variates[,i],metaboloma[,j],method = "spearman")$p.value
  }
    
}
colnames(matriz) <- colnames(metaboloma)
colnames(matriz.p)<- colnames(metaboloma)
corrplot::corrplot(matriz,is.corr = F)

```
```{r}
csa1 <- colnames(metaboloma)[which(p.adjust(matriz.p[1,],method = "bonferroni")<0.05)]
csa2 <- colnames(metaboloma)[which(p.adjust(matriz.p[1,],method = "bonferroni")<0.05)]
metabolitos <-unique(c(csa1,csa2))
metabolitos
```

```{r}
library(lmPerm)
p<-vector(mode="numeric",length=length(metabolitos))
for(i in 1:length(metabolitos)){
  metabolito <- metabolitos[i]
  s <- summary(aovp(metaboloma[,metabolito]~grupo*obesidad))[[1]]
  p[i]<- s$`Pr(Prob)`[3]
  
}
names(p) <- metabolitos
p
```
```{r}
p[p.adjust(p,method="BH")<0.05]

```
```{r}
pairwise.wilcox.test(metaboloma$Lysine_mean,interaccion,p.adjust.method = "BH")

```
```{r}
pairwise.wilcox.test(metaboloma$Formate_mean,interaccion,p.adjust.method = "BH")
```

```{r}
pairwise.wilcox.test(metaboloma$Alanine_mean,interaccion,p.adjust.method = "BH")

```
## Metagenoma



```{r}
library(zCompositions)
genus.abs<-genus[,-grep("^ud-",colnames(genus))]

genus.abs[genus.abs==0]<-genus.abs[genus.abs==0]+2
genus.imput <-cmultRepl(X = genus.abs,label = 1,output = "prop")

```
```{r}
library(mixOmics)
library(reshape2)
library(vegan)
G <-matrix(t(logratio.transfo(t(genus.imput),logratio = "CLR")),ncol=ncol(genus.abs),nrow(genus.abs))

# datos <- t(rlog(counts(dds.interaccion.nb.2)))
datos = as.data.frame((G))
p1<-fn.densidad(datos = datos,fctr = interaccion)
p2 <- fn.boxplot(datos = datos,fctr = interaccion)
pca.log <- prcomp((datos),center = T,scale. = F)
pca.var <- 100*(pca.log$sdev^2/sum(pca.log$sdev^2))
p3<-graphPCA.methods(pca.log$x[,1:2],
                     datos.common = datos.common,
                     varianzas=pca.var)+ggtitle("PCA")

mds <- cmdscale(vegdist(genus.imput,method = "aitchison"),eig = T,add = F,k=10)
varianzas <- 100*mds$eig/sum(mds$eig)
pca.spearman <- princomp(mds$points,fix_sign = F,cor=T)

pca.spearman.scores <- pca.spearman$scores
pca.spearman.var <- 100*(pca.spearman$sdev^2/sum(pca.spearman$sdev^2))
p4<-graphPCA.methods(pca.spearman.scores[,1:2],
                 datos.common = datos.common,
                 varianzas = pca.spearman.var)+ggtitle("PCA spearman")



mds <- cmdscale(vegdist(genus.imput,method = "aitchison"),eig = T,add = F,k=10)
varianzas <- 100*mds$eig/sum(mds$eig)
p5<-graphPCA.methods(datos=mds$points,
                     datos.common = datos.common,
                     varianzas = varianzas)+ggtitle("PCoA")


p5<-graphPCA.methods(datos=mds$points,
                     datos.common = datos.common,
                     varianzas = varianzas)+ggtitle("PCoA")



p11<-ggarrange(p1,p2,p3,p5,common.legend = T)
annotate_figure(p11,bottom = "media ajusta y rlog")


```
```{r}
adonis2(genus.imput~obesidad*grupo,method="robust.aitchison")
```


```{r}
library(ALDEx2)
set.seed(123)
conds <- grupo
x.all <- aldex.clr(t(genus.abs),conds)
mc.instances <- numMCInstances(x.all)
mc.all <- getMonteCarloInstances(x.all)
feature.number <- numFeatures(x.all)

p.grupo <- vector("numeric",length=mc.instances)
p.obesidad <- vector("numeric",length=mc.instances)
p.interaccion <- vector("numeric",length=mc.instances)

p.grupoBH <- vector("numeric",length=mc.instances)
p.obesidadBH <- vector("numeric",length=mc.instances)
p.interaccionBH <- vector("numeric",length=mc.instances)
  for (mc.i in 1:mc.instances) {
    t.input <- t(sapply(mc.all, function(y) {
      y[, mc.i]
    }))
    
    permanova <- adonis2(t.input~obesidad*grupo,method = "euclidean")
    p.obesidad[mc.i] <- permanova$`Pr(>F)`[1]
    p.grupo[mc.i]  <- permanova$`Pr(>F)`[2]
    p.interaccion[mc.i]  <- permanova$`Pr(>F)`[3]
    # wi.p.matrix[, mc.i] <- wilcox.fast(t.input, setAsBinary, 
    #   paired.test)
  }

acc.obesidad <- vector("numeric",length=mc.instances)
curiosidad <- NULL

  for (mc.i in 1:mc.instances) {
    t.input <- t(sapply(mc.all, function(y) {
      y[, mc.i]
    }))
    
    plsda.obesidad <- mixOmics::plsda(t.input,grupo,ncomp = 2)
    prediction <- mixOmics::predict(plsda.obesidad,(t.input))
    prediction <- as.factor(prediction$MajorityVote$max.dist[,1])
    acc.obesidad[mc.i] <- caret::confusionMatrix(prediction,grupo)$overall
    actual <- c(female=length(which(grupo=="Female")),
                male=length(which(grupo=="Male")),
                pcos=length(which(grupo=="PCOS")))
    prediction. <- c(female=length(which(prediction=="Female")),
                male=length(which(prediction=="Male")),
                pcos=length(which(prediction=="PCOS")))
    
    # wi.p.matrix[, mc.i] <- wilcox.fast(t.input, setAsBinary, 
    #   paired.test)
  }
which.max(acc.obesidad)

t.input <- t(sapply(mc.all, function(y) {
      y[, which.max(acc.obesidad)]
    }))
plsda.obesidad <- mixOmics::plsda(t.input,obesidad)
p5<-graphPCA.methods(plsda.obesidad$variates$X,datos.common,varianzas=100*c(plsda.obesidad$prop_expl_var$X))+ggtitle("PLS-DA")
p5
```
```{r}
plsda.obesidad$prop_expl_var$Y
```

```{r}
x.all <- aldex.clr(t(genus.abs),obesidad)
kruskalaldex <-aldex.ttest(x.all)
```


```{r}
plsda.group <- plsda.obesidad
plsa.out <- plsda.group$variates$X %*% t(plsda.group$loadings$X)
```



```{r}
adonis2(plsa.out~obesidad*grupo,method = "euclidean")
```

# RESULTADOS MOFA

# INTEGRACION

```{r}
library(MOFA2)
unique(strsplit2(list.files("modelsMOFA/"),"_")[,3])

models <- lapply(list.files("modelsMOFA/",full.names = T), load_model)

elbos <-compare_elbo(models,return_data = T)

model <-load_model(list.files("modelsMOFA",full.names = T)[which.max(elbos$ELBO)])
```

### Generalidades

```{r}
plot_factor_cor(model)

```
```{r}
ggarrange(plot_variance_explained(model, x="view", y="factor"),
plot_variance_explained(model, y="factor", plot_total = T)[[2]])
calculate_variance_explained(model)
```


```{r}
correlate_factors_with_covariates(model, 
  covariates = c("obesidad","grupo","edad","BMI","WC","WHR","SHGB","hsCRP","TE2","E2"), 
  plot="log_pval"
)
```


```{r}
model@covariates <-data.frame(obesidad=as.numeric(obesidad),grupo=as.numeric(grupo),edad=Clin$EDAD,BMI=Clin$BMI,WC=Clin$WC,WHR=Clin$WHR,SHGB=Clin$SHBG,hsCRP=Clin$hsCRP,TE2=Clin$Total_ESTR,E2=Clin$Free_ESTRA)

```



```{r}
factors <- get_factors(model, as.data.frame = T,scale = F)
plot.methods <- function(factores,datos.common){
  
  
  sujetos <-data.frame(Sujetos=1:length(factores),Factor=factores)
  sexo<-datos.common$GROUP
  obesidad<-datos.common$OBESE
  p1<-ggplot(sujetos,aes(x=Sujetos, y=Factor)) +
     geom_hline(yintercept = 0, color = "gray70") +
     geom_vline(xintercept = 0, color = "gray70") +
     geom_point(aes(color = sexo,shape=obesidad,fill=obesidad), alpha = 0.55, size = 3)  +
     scale_color_manual(values=c("black","red","green"))+scale_shape_manual(values=c(17,15))
  return(p1)
}
factores.recast <-as.data.frame(reshape2::recast(data=factors,sample~factor))
rownames(factores.recast) <- factores.recast$sample
factores.recast <- factores.recast[,-1]
var.explained <-calculate_variance_explained(model)
fac1 <- plot.methods(factores.recast[,1],datos.common)+ ggtitle(
  paste("Factor 1 \n Metaboloma ",round(var.explained$r2_per_factor$group1[1,1],2),"\n Microbioma - ",round(var.explained$r2_per_factor$group1[1,2],2)))
                                                               
                                                               
fac3 <- plot.methods(factores.recast[,2],datos.common)+ggtitle(paste("Factor 4","\n Metaboloma - ",round(var.explained$r2_per_factor$group1[2,1],2), "\n Microbioma - ",round(var.explained$r2_per_factor$group1[2,2],2)))

fac4 <- plot.methods(factores.recast[,4],datos.common)+ggtitle(paste("Factor 4","\n Metaboloma - ",round(var.explained$r2_per_factor$group1[4,1],2), "\n Microbioma - ",round(var.explained$r2_per_factor$group1[4,2],2)))

ggarrange(fac1,fac3,fac4,common.legend = T)
```


### DIMRED

```{r}

factors <- get_factors(model, as.data.frame = T,scale = T)

weights <- rbind(get_weights(model,as.data.frame=T,scale=T,views="metaboloma"),
                 get_weights(model,as.data.frame=T,scale=T,views="microbioma"))
                 
factores.recast <-as.data.frame(reshape2::recast(data=factors,sample~factor))
rownames(factores.recast) <- factores.recast$sample
factores.recast <- factores.recast
weights.recast <-as.data.frame(reshape2::recast(data=weights,feature~factor))
rownames(weights.recast) <- weights.recast$feature
weights.recast <- weights.recast
weights.recast$views <- c(rep("metabooloma",37),rep("microbioma",213))
Y <-(as.matrix(factores.recast[,-1]) %*% t(weights.recast[,-c(1,ncol(weights.recast))]))



```

```{r}
## Obese
pcy <- prcomp((Y),scale. = F)
varianzas <- 100*(pcy$sdev^2/sum(pcy$sdev^2))
datos <- as.data.frame(bind_cols(pcy$x[,1:2]))
graphPCA.methods(datos,datos.common = datos.common,varianzas)
```




# COSAS QUE HACER
## METABOLOMA SIN PROCESADO
## PLSDA Y PCA
## REDUCCION SIN 213
## REDUCCION CON 213
##

# ```{r}
# library(MOFA2)
# set.seed(123)
# 
# seed <- sample(1000,10)
# mc.instances <- numMCInstances(x.all)
# mc.all <- getMonteCarloInstances(x.all)
# i<-0
# X<-(voom(met37_norm,normalize.method = "quantile",design = design)$E)
# for(s in seed){
# for(mc.i in 1:mc.instances) {
#   i<-i+1
#     t.input <- (sapply(mc.all, function(y) {
#       y[, mc.i]
#     }))
# 
#     genero <- t.input
#     datos <- list(metaboloma=(X),microbioma=(genero))
#     MOFAobject <-create_mofa_from_matrix(datos,groups=NULL)
#     samples_metadata(MOFAobject)<- data.frame(sample=pacientes,obesidad=(obesidad),grupo=(grupo),edad=Clin$EDAD,BMI=Clin$BMI,WC=Clin$WC,WHR=Clin$WHR,SHGB=Clin$SHBG,hsCRP=Clin$hsCRP,TE2=Clin$Total_ESTR,E2=Clin$Free_ESTRA,obesidad=as.numeric(obesidad),grupo=as.numeric(grupo))
#   rownames(samples_metadata(MOFAobject))<-pacientes
#   data_opts <- get_default_data_options(MOFAobject)
#   data_opts$scale_views <-T
#   model_opts <- get_default_model_options(MOFAobject)
#   model_opts$likelihoods=c("gaussian","gaussian")
#   model_opts$num_factors=10
#   head(model_opts)
#   model_opts$spikeslab_factors<-F
#   model_opts$ard_factors <-T
#   train_opts <- get_default_training_options(MOFAobject)
#   train_opts$seed <- s
# # outfile = file.path(getwd(),paste0(seq(1:10),"_model.hdf5"))
#   MOFAobject <- prepare_mofa(
#   object = MOFAobject,
#   data_options = data_opts,
#   model_options = model_opts,
#   training_options = train_opts)
#     outfile <- file.path(getwd(),"modelsMOFA/",paste0("model_","seed_",s,"_instace_",mc.i,".hdf5"))
#   reticulate::use_python("/usr/bin/python3.10")
# 
#   MOFAobject <- run_mofa(MOFAobject, outfile,use_basilisk = F)
#   MOFAobject@covariates<-data.frame(obesidad=as.numeric(obesidad),grupo=as.numeric(grupo),edad=Clin$EDAD,BMI=Clin$BMI,WC=Clin$WC,WHR=Clin$WHR,SHGB=Clin$SHBG,hsCRP=Clin$hsCRP,TE2=Clin$Total_ESTR,E2=Clin$Free_ESTRA)
# 
# }
# }
# ```



