---
title: Integration of omics and clinical data of the hormonal, metabolic, inflammatory,
  and oxidative response in the different macronutrients of the diet
author: "Edmond Geraud"
subtitle: '`r params$subtitulo`'
date: '`r format(Sys.Date(),"%e de %B, %Y")`' 
output:
  html_document:
    toc: true
    toc_depth: 2
  prettydoc::html_pretty:
    theme: cayman
    highlight: github
    toc: true
header-includes:
  - \usepackage[english]{babel}
params:
  file.data: Integromics_IPmarkers.xlsx
  folder.data: ../../datos
  subtitulo: Statistical Descriptive analysis of Intestinal Permeability Biomarkers
geometry: margin=2cm
---

```{r setup, include=FALSE}
require(knitr)
# include this code chunk as-is to set options
opts_chunk$set(comment = NA, prompt = TRUE, tidy = T, 
               fig.width = 7, fig.height = 7,echo = TRUE, 
               message = FALSE, warning = FALSE, cache=TRUE, fig_caption = TRUE)
Sys.setlocale("LC_TIME", "C")
```


```{r libraries}

list.of.packages <- c("xlsx","kableExtra","dplyr","ggplot2","egg","cowplot","patchwork","gridExtra","UsingR","car","lattice","ggpubr","ggbreak","GGally","reshape2","ggcorrplot","corrplot","rela","ggrepel","factoextra")
new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[,"Package"])] ## check for packages not installed 
if(length(new.packages)) install.packages(new.packages) ## install packages if necessary
res<-unlist(lapply(list.of.packages, require,character.only = T)) ## load packages needed for the session
if(any(res==F)){
  list.of.packages[which(res==F)]  ## show those package if you have troubles to install.
}

list.of.bioc.packages <- c("phyloseq","pcaMethods")
new.packages.bioc <- list.of.packages[!(list.of.bioc.packages %in% installed.packages()[,"Package"])]
if(length(new.packages.bioc)) BiocManager::install(new.packages.bioc)
res.bioc <- unlist(lapply(list.of.bioc.packages, require,character.only = T)) 
if(any(res.bioc==F)){
  list.of.packages[which(res.bioc==F)]  ## show those package if you have troubles to install.
}
```


```{r funciones}
resumen<-function(x){c(round(mean(x),4),round(sd(x),4))} # Function that help us to summarize the data in mean and sd values
lowerFn <- function(data, mapping, method = "lm", ...) {
  p <- ggplot(data = data, mapping = mapping) +
    geom_point(colour = "blue") +
    geom_smooth(method = method, color = "red", ...)
  p
}

## Function that takes variable names and the labels for the axis.
bp.1.vars<-function(datos_,vars,y_lab){
  
  datos <- cbind(datos_[,c("GROUP","OBESE")],datos_[,vars])
  df.melt<-melt(datos,id.vars=c("GROUP","OBESE"))
  
    # print(head(df.melt))


  p2<-ggplot(df.melt,aes(y=value,x=variable,fill=GROUP))+geom_boxplot()+ylab(y_lab)+facet_grid(~OBESE)+xlab("") +geom_hline(yintercept=0, linetype="dashed", color = "red")+theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=1,color = "black",size=7))
  
  
  return(p2)
  
}

plotPCA_scores <- function (X,factor, title, size = 1.5, glineas = 0.25,center_,scale_,labels) {

    pcX <- prcomp(X,center=center_,scale. = scale_)
    X.scores <- pcX$x
    rot <- round(pcX$sdev^2/sum(pcX$sdev^2)*100,2)

  # scores
  X.scores <- as.data.frame(X.scores)
  Group <- factor[[1]]
  Group2<-factor[[2]]
  colores<-1:length(levels(Group))
   # main plot
   p1 <- ggplot(X.scores,aes(x=PC1, y=PC2)) +
     theme_classic() +
     geom_hline(yintercept = 0, color = "gray70") +
     geom_vline(xintercept = 0, color = "gray70") +
     geom_point(aes(color = Group,shape=Group2), alpha = 0.55, size = 3) +
     coord_cartesian(xlim = c(min(X.scores[,1])-5,max(X.scores[,1])+5)) +
     scale_fill_discrete(name = "Group")
   # avoiding labels superposition
   
      grafico<- p1 + geom_text_repel(aes(y = PC2 + 0.25, label = labels),segment.size = 0.25, size = size) + 
     labs(x = c(paste("PC1",rot[1],"%")),y=c(paste("PC2",rot[2],"%"))) +  
     ggtitle(paste("Principal Component Analysis for: ",title,sep=" "))+ 
     theme(plot.title = element_text(hjust = 0.5)) +
     scale_color_manual(values=colores)
     
 
     

return(list(pcX,grafico))}
```


# Summary of the univariate descrpitve analysis.

## Overall study.

The data comprises 53 subjects, which are divided into 34 females and 19 males. From the 34 females, 18 are non obese and 16 obese. And 10 non obese males and 9 obese males. The 18 non obese females are divided into 9 PCOS and 9 healthy. The 16 obese females are divided further into 8 PCOS and 8 non PCOS.

A sanity check of the study was performed to observed possibly differences between the groups. No differences were found.
Descriptive numerical tables were performed on each data-set except on the microbiome. Numerical descriptive tables only contain the mean and the standard deviation.

For each numerical variable (all except the ones that describes the groups) their distribution were analyzed as a whole, without taking into account the groups (obese, sex and PCOS).

It has to be mentioned that no correlation between variables on the different data types were performed. Only uni variate statistical analysis on each data set was performed.

Permeability markers, and some clinical biomarkers, were analyzed at basal and postprandial times. The first day of the study corresponds to the glucose oral load, the second to the lipids and the third to the proteins.


**Permability Biomarkers:** Apart from the categorical variables, we have sCD14, LBP, zonulin and succinate. Each variable the mean basal value is computed,as well as the basal values for each day and the postprandial stage. Succinate is more like a metabolite, however, it was calculated in other units, and we do not know how to include this metablolite. ( No units are specified on this data set)


## Intestinal permeability markers. 

As mentioned above, this data set comprises of four permeability markers and one metabolite. As we did on the clinical biomarkers, firstly we did a quick review of the literature.

CD14 (Cluster differentiation 14), is a protein that detects bacteria in the body by binding a lipopolysacharide *This should be interesting when integrating the data sets*. Also this binding its only possible with the *Lipopolysacharide-binding-protein LBP*. Also recognize other pathogen associated molecular patterns. LBP is important to cell surface pattern recongition such as CD14 and TLR4 (mentioned on the references).

GLP2 (Glucagon-like peptide2) decreases cytokine expression, and is part of the inflammatory response.
Is reported than Zonulin increases in patients with diabetes mellitus II. 

On the other hand, succinate, as mentioned, is a metabolite, involved in many pathways, activates the inflammatory response trough toll like receptors (mentioned on the main references).

A correlation-scatter matrix plot was performed to observe the relationships among the mean basal values of the variables, BMI included, as is an indicator of diabetes II and inflammation.

BMI, correlates positively with all variables, except on sCD14. 
SCD14, correlates in a positve manner with LBP and Zonulin, as described in the literature, and negatively with GLP2 and succinate. 

The correlation of LBP with GLP2, zonulin and succinate is positive. 
There is no relationship between GLP2 and zonulin, but a positive correlation exist between succinate. Finally succinate and zonulin present a barely positive correlation.


**Comparison between basal and postpranidal levels.**

All variables on this data set, show higher values at basal level than at postprandial time
SCD14,  at basal levels is higher on non-obese subjects. Specially on healthy control women. No distinction between the other groups is found regarding non-obese subjects. Obese PCOS women show increased levels of this IP, but lower than healthy control women. At postprandial levels, similar values are found across all subjects. But on the second and third day, on non-obese subjects males and PCOS women show higher values. Obese males at the first day show the highest values at postprandial levels.

LBP mean basal values are higher on obese subjects, specifically on PCOS obese women. And no distinction between groups are observed at postprandial levels, GLP values are higher at basal levels on obese PCOS women and obese males. Between control women no distinction is observed. At postprandial levels,  higher values are observed on the second and third day. With no distinction between non-obese subjects. Obese control subjects show higher values than obese PCOS women.

Zonullin mean basal levels show no distinction between groups. At postprandial levels, on the second day, between obese obese subjects, control females have the highest values followed by PCOS and males.

Succinate levels are higher on obese subjects, but remarkably higher on obese women with PCOS. At postprandial time, the first day, levels of this metabolite are higher. Being higher on non obese males and obese PCOS women. On the other days no distinction is found between groups


## Description of Intestinal Permeability Markers.

Read the file of IP
```{r IPmarkerssData}
IP_file<-file.path(params$folder.data,file=params$file.data) ## file path of the data
IP.tmp<-read.xlsx(IP_file,sheetIndex = 1)
```

Observe the structure of the data
```{r chunkip1}
str(IP.tmp) # show general description of the data
```

We observe form the information displayed that there is a column of missing values. Furthermore, we have the same number of observations, 57, and 42 variables. Form these variables, 35 variables are new. 5 variables we have already have it from the previous analysis, and the first one is de order of the samples. Thus, in order to proceed we have to remove the missing value column. This column is only for visualization purposes in Excel. 

We could merge the variables form the first dataset, but to be sure that the order of samples is not another, weare going to perform a sanity check. Furthermore, as in the clinical dataset, the categorical variables are in `character` class. So we need also to transform into factors. 


```{r}
IP<-IP.tmp[,-7] # remove the missin values column
IP$SEX <- factor(IP$SEX,levels=c(0,2),labels = c("Female","Male"))
IP$OBESE <- factor(IP$OBESE,levels=c(0,1),labels = c("No Obese","Obese"))
IP$GROUP <- factor(IP$GROUP,levels=c(0,1,2),labels = c("Female","PCOS","Male"))
# all(IP$Paciente == general_data$Paciente) # sanity check.
# str(IP)
```

The data comprises the following permeability markers.

1. CD14

Cluster differentiation 14 (CD14), is a human protein  made mostly by macrophages as part of the innate immune system. Detects **bacteria** in the body by  binding **lipopolysaccharide** (LPS), a pathogen-associated molecular pattern. It is anchored to the membrane by a **glycosylphosphatidylinositol** (GPI) CD14 acts as a co-receptor (along with the *Toll-like receptor TLR 4 and MD-2*) for the detection of bacterial lipopolysaccharide (LPS). CD14 can bind LPS only in the presence of **lipopolysaccharide-binding protein (LBP)** but also recognizes other pathogen-associated molecular patterns such as *lipoteichoic acid* soluble

2. LBP

Lipopolysaccharide binding protein: LBP is a soluble acute-phase protein that binds to bacterial lipopolysaccharide (or LPS) to elicit immune responses by presenting the LPS to important cell surface pattern recognition receptors called CD14 and TLR4. The protein encoded by this gene is involved in the acute-phase immunologic response to gram-negative bacterial infections. 
The protein encoded by this gene is involved in the acute-phase immunologic response to gram-negative bacterial infections. The proinflammatory activity of plasma LPS is increased by LPB, which is higher in obesity Lipopolysaccharide-binding protein has been shown to interact with CD14, TLR2, TLR4 and the co-receptor MD-2


3. GLP2


Glucagon-like peptide 2 (GLP-2) is a 33 amino acid peptide-encoded carboxyterminal to the sequence of GLP-1 in the proglucagon gene. Both GLP-1 and GLP-2 are secreted from gut endocrine cells and promote nutrient absorption through distinct mechanisms of action. GLP-2 regulates gastric motility, gastric acid secretion, intestinal hexose transport, and increases the barrier function of the gut epithelium. GLP-2 reduces mortality and decreases mucosal injury, cytokine expression, and bacterial septicemia in the setting of small and large bowel inflammation. 



4. Zonulin

Zonulin (haptoglobin 2 precursor) is a protein that modulates the permeability of tight junctions between cells of the wall of the digestive tract.  Type 2 diabetic patients have shown increased zonulin


5. Succinate

Activation of inflammatory macrophages through toll-like receptors induces a metabolic shift towards glycolysis.....Actually is a metabolite 

For each intestinal permeability marker, it is measured at basal stage and mean, for each day corresponding to every marcronutrient load. For each day they only measured values at basal time and at 60 minutes or 120 minutes, depending if it the load was lipids or not.


### Description of the variables 

| Variable 	| Description 	|
|:---:	|:---:	|
| Orden 	| Order of the subjects 	|
| Paciente 	| Women contol (M),PCOS women (P),Male (V), d (Not obese), O (Obese), GLP (challenge macronutrients: glucose, lipids, proteins) 	|
|  	<br>SEX 	| 0 Women, 2 Male 	|
|  	<br>GROUP 	| 0 Women Control, 1 PCOS, 2 Male 	|
|  	<br>OBESE 	| 0 Not obese, 1 Obese 	|
| BMI 	| Body Mass Index 	|
| CD14_basalmean 	| Mean basal concentration: Cluster differentiation 14 (CD14) 	|
| G_CD14_0 	| Basal time Glucose load day: Cluster differentiation 14 	|
| L_CD14_0 	| Basal time Lipid load day: Cluster differentiation 14 	|
| P_CD14_0 	| Basal time Protein load day: Cluster differentiation 14 	|
| AUC_CD14_Glu 	| Postprandial time Area Under the Curve Glucose load day: Cluster differentiation 14 	|
| AUC_CD14_Lipids 	| Postprandial time Area Under the Curve Lipids load day: Cluster differentiation 14 	|
| AUC_CD14_Prots 	| Postprandial time Area Under the Curve Rroteins load day: Cluster differentiation 14 	|
| LBP_basalmean 	| Mean basal concentration:  Lipopolysaccharide binding protein  	|
| G_LBP__0 	| Basal time Glucose load day: Lipopolysaccharide binding protein 	|
| L_LBP_0 	| Basal time Lipid load day: Lipopolysaccharide binding protein 	|
| P_LBP_0 	| Basal time Protein load day: Lipopolysaccharide binding protein 	|
| AUC_LBP_Glu 	| Postprandial time Area Under the Curve Glucose load day: Lipopolysaccharide binding protein 	|
| AUC_LBP_Lipids 	| Postprandial time Area Under the Curve Lipids load day: Lipopolysaccharide binding protein 	|
| AUC_LBP_Prots 	| Postprandial time Area Under the Curve Proteins load day: Lipopolysaccharide binding protein 	|
| GLP2_basalmean 	| Mean basal concentration: Glucagon-like peptide 2 	|
| G_GLP2_0 	| Basal time Glucose load day: Glucagon-like peptide 2 	|
| L_GLP2_0 	| Basal time Lipid load day: Glucagon-like peptide 2 	|
| P_GLP2_0 	| Basal time Protein load day: Glucagon-like peptide 2 	|
| AUC_GLP2_Glu 	| Postprandial time Area Under the Curve Glucose load day: Glucagon-like peptide 2 	|
| AUC_GLP2_Lipids 	| Postprandial time Area Under the Curve Lipids load day: Glucagon-like peptide 2 	|
| AUC_GLP2_Prots 	| Postprandial time Area Under the Curve Proteins load day: Glucagon-like peptide 2 	|
| Zonulin_basalmean 	| Mean basal concentration: zonulin 	|
| G_Zonulin_0 	| Basal time Glucose load day: zonulin 	|
| L_Zonulin_0 	| Basal time Lipid load day: zonulin 	|
| P_Zonulin_0 	| Basal time Protein load day: zonulin 	|
| AUC_Zonulin_Glu 	| Postprandial time Area Under the Curve Glucose load day: zonulin 	|
| AUC_Zonulin_Lipids 	| Postprandial time Area Under the Curve Lipids load day: zonulin 	|
| AUC_Zonulin_Prots 	| Postprandial time Area Under the Curve Rroteins load day: zonulin 	|
| Succin_basalmean 	| Mean basal concentration: Succinate* 	|
| G_Succinate_0 	| Basal time Glucose load day: Succinate* 	|
| L_Succinate_0 	| Basal time Lipid load day: Succinate 	|
| P_Succinate_0 	| Basal time Protein load day: Succinate* 	|
| AUC_Succ_Glu 	| Postprandial time Area Under the Curve Glucose load day: Succinate* 	|
| AUC_Succ_Lipids 	| Postprandial time Area Under the Curve Lipids load day: Succinate* 	|
| AUC_Succ_Prots 	| Postprandial time Area Under the Curve Rroteins load day: Succinate* 	|

### Sanity check: Statistical description of categorical variables.


```{r intialtables2}
(sexo<-table(IP$SEX))

(sexo.obesos<-table(IP$SEX,IP$OBESE))

(subgrupos<-table(IP$GROUP,IP$OBESE))
```

```{r plot1_IP, fig.align='center'}
ggplot(data = IP,aes(x=GROUP,y=OBESE,fill=OBESE)) + geom_bar(stat = "identity",color="black",position = position_dodge())+theme_minimal()

```

### Univariate descriptive analysis of IP markers

#### Numerical descriptive analysis

Before starting with the graphical statistical analysis, we are going to perform a numerical descriptive analysis.

```{r chunkIPdes}

IP.variables <- IP[,3:ncol(IP)] ## remove samples names and order from the summary adn BMI variable, we have already described



IP.sum<-IP.variables %>%group_by(GROUP,OBESE) %>% 
  summarise(across(where(is.numeric),resumen),n=n()) # Group the data by variable GROUP and OBESE

d.f<-as.data.frame(t(IP.sum))[c(nrow(t(IP.sum)),1:(nrow(t(IP.sum))-1)),] # transpose the data for visualization purposes and get rid off the N value for same purposes
d.f_final <- d.f[-c(1:4),] # remove categorical variables.
nombre_variables<- c("Mean basal value sCD14",
                  "Basal gluclose load sCD14",
                  "Basal lipid load sCD14",
                  "Basal protein load sCD14",
                  "Postprandial AUC glucose sCD14",
                  "Postprandial AUC lipid sCD14",
                  "Postptrandial AUC protien sCD14",
                   "Mean basal value LBP",
                  "Basal gluclose load LBP",
                  "Basal lipid load LBP",
                  "Basal protein load LBP",
                  "Postprandial AUC glucose LBP",
                  "Postprandial AUC lipid LBP",
                  "Postptrandial AUC protien LBP",
                  "Mean basal value GLP2",
                  "Basal gluclose load GLP2",
                  "Basal lipid load GLP2",
                  "Basal protein load GLP2",
                  "Postprandial AUC glucose GLP2",
                  "Postprandial AUC lipid GLP2",
                  "Postptrandial AUC protien GLP2",
                   "Mean basal value zonulin",
                  "Basal gluclose load zonulin",
                  "Basal lipid load zonulin",
                  "Basal protein load zonulin",
                  "Postprandial AUC glucose zonulin",
                  "Postprandial AUC lipid zonulin",
                  "Postptrandial AUC protien zonulin",
                   "Mean basal value Succinate",
                  "Basal gluclose load Succinate",
                  "Basal lipid load Succinate",
                  "Basal protein load Succinate",
                  "Postprandial AUC glucose Succinate",
                  "Postprandial AUC lipid Succinate",
                  "Postptrandial AUC protien Succinate")
code_variables <- colnames(IP)[7:ncol(IP)]
d.f_final$Coda <- code_variables
rownames(d.f_final) <-nombre_variables
N<-as.character(d.f_final[1,])
colnames(d.f_final) <- rep(c("Mean","SD"),6)
## ATTENION! Because assining left hand expression is 
## complicated, just for one table, be careful if we put another data-set
## For showing the number of subjects by group
kbl(d.f_final) %>%
  kable_classic_2() %>%
    add_header_above(c(" "=1,
                      "9"= 2,
                      "8"= 2,
                      "9"= 2,
                      "8"= 2,
                      "10"= 2,
                      "9"= 2,
                      " "=1
 )) %>%
      add_header_above(c( " "=1,
                     "No Obese" = 2,
                     "Obese" =2,
                     "No Obese" = 2,
                     "Obese" =2,
                     "No Obese" = 2,
                     "Obese" =2, 
                     " "=1)) %>%
  add_header_above(c(" "=1,
                      "Control Females" = 4, 
                     "PCOS"=4,
                     "Control Males"=4,
                     " "=1))



```


#### General graphical descriptive analysis of the mean values at basal stage.

Like we have doing until now, firstly let's describe the distribution of the main variables: sCD14, LBP, zonulin, and succinate. We hace to take into account that succinate is more like a metabolite rather than an intenstinal permeability marker. Furhtemore, **there is the doubt** if we have to analyze this metabolite with the metabolome or with this dataset, because the units are differents.

**WHAT ARE THE UNITS OF THIS DATA SET ?**


```{r chunkIP4}

variables <- colnames(IP)[7:ncol(IP)]
hist.list.IP<-list() ## initizalize a list of the histogram
## plot all histograms via loop
for(i in 1:length(variables)){
  variable <- variables[i] ## name of the variable to plot
  titulo <- nombre_variables[i] # title of the varialbe
  
  breaks <- pretty(range(IP[,variable]), 
                   n = nclass.Sturges(IP[,variable]), min.n = 1)# make pretty histogram by appropiate bins
  bwidth <- breaks[2]-breaks[1] ## bin width

  ## plot
  hist.list.IP[[i]]<-ggplot(IP,aes_string(x=variable))+geom_histogram(aes(y=..density..),
                                                        binwidth =bwidth,colour="black")+geom_density(alpha=.2, fill="lightblue")+xlab(titulo)+ylab("")


}


bp.interaction <- list() ## intialize boxplot list interaction
GROUP <- IP$GROUP
OBESE<- IP$OBESE
for(i in 1:length(variables)){
  titulo <- nombre_variables[i]
  variable <- variables[i]
  bp.interaction[[i]]<-ggplot(IP,aes_string(x=GROUP,y=variable,interaction(GROUP),fill=OBESE))+geom_boxplot()+ylab(titulo)

}
```


```{r}
do.call(grid.arrange,hist.list.IP[grep("basalmean",colnames(IP)[7:ncol(IP)])])
```
All mean basal values for the variables in this dataset follows a right skewed distribution, except the mean basal value of `LBP`.

Now let's take a look at the main relationships between them and the BMI.

```{r}
ggpairs(
  IP[,c(6,grep("basalmean",colnames(IP)))], lower = list(continuous = wrap(lowerFn, method = "lm")),
  diag = list(continuous = wrap("barDiag", colour = "blue")),
  upper = list(continuous = wrap("cor", size = 5)),
  columnLabels = c("BMI","sCD14","LBP","GLP2","zonulin","Succinate")
)
```

In general terms, without making groups, BMI correlates positively with all basal levels of `LBP`, `GLP2`, `zonulin` and `Succinate`. This means, while BMI augments, all levels increases except `CD14` in which the correlation is negative. All correlations are statistically significant. 

Regarding, `CD14` none of the correlation is statistically significant, furthermore, barely is there any correlation. Between positive with LBP and `zonulin` and negative for `Succinate` and `GLP2`.

`LBP` correlates statistical significant in a positive manner with `zonulin` and `Succinate`. Positive also, but statistically significant with `GLP2`

`GLP2` with `zonulin` though the data tells us there is a positive correlation, practically this correlation does not exist. Though is statistical significant positive with `Succinate`. Furthermore `Succinate` and `zonulin` does not correlate. 


Now, we are going to proceed to observe the different values of each variable at basal, Postprandial and between the different days.

##### CD14

```{r cd14}

# cd14_basal<-bp.interaction[1]
p1<-bp.1.vars(IP,c("G_CD14_0","L_CD14_0","P_CD14_0"),"Basal CD14")
p2<-bp.1.vars(IP,c("AUC_CD14_Glu","AUC_CD14_Lipids","AUC_CD14_Prots"),"Postpradial CD14")
grid.arrange(p1,p2)
```

Postpradial levels of `CD14` are lower than basal levels of this permeability marker. Non-obese women have higher values of `CD14` at basal stage, followed by non-obese men. Obese PCOS women, have the highest values between obese subjects, but lower than healthy women, and similar values in non-obese PCOS women. Obese men and obese women, have lower levels than non-obese subjects. Postpradial levels, are lower in non-obese subjects, being lower women. Obese men at the first day have the highest values.

##### LBP
```{r}
# lbp_basal<-bp.interaction[8][[1]]
p1<-bp.1.vars(IP,c("G_LBP__0","L_LBP_0","P_LBP_0"),"Basal LBP")
p2<-bp.1.vars(IP,c("AUC_LBP_Glu","AUC_LBP_Lipids","AUC_LBP_Prots"),"Postpradial LBP")+ theme(axis.text.x = element_text(angle = 0, vjust = 0.9, hjust=1,size = 7))
grid.arrange(p1,p2)
```

Basal levels of `LBP` are higher than Postpradial stage. The IP data tells us that the mean basal values are higher in obese subjects, but mostly on PCOS obese women. At Postpradial stage, higher values have the non.obese subjects, but with more variability in obese subjects. Obese men at the lipid load day, have the highest values at Postprandial levels. In general at Postpradial stage the levels are similar between non-obese subjects.

##### GLP


```{r glp}
lbp_basal<-bp.interaction[15][[1]]
p1<-bp.1.vars(IP,c("G_GLP2_0","L_GLP2_0","P_GLP2_0"),"Basal GLP2")
p2<-bp.1.vars(IP,c("AUC_GLP2_Glu","AUC_GLP2_Lipids","AUC_GLP2_Prots"),"Postpradial GLP2")+ theme(axis.text.x = element_text(angle = 0, vjust = 0.9, hjust=1,size = 7))
grid.arrange(p1,p2)
```

Again, basal levels of `GLP2` are higher than at Postpradial stage. Regarding the mean basal values, we observe that the obese PCOS and men have higher values. Control females, both obese and non-obese have similar distribution of values. While healthy males and PCOS obese women have similar values. If we see the data at the different days, on non-obese PCOS women, these patients have the lower values.

At Postpradial time, in general, across subjects, the valuesof `GLP2` increases at the lipid and protein levels progressively. With similar values between subjects on non-obese persons. Obese PCOS women, show lower values of `GLP2`.



##### zonulin


```{r zonulin}
lbp_basal<-bp.interaction[22][[1]]
p1<-bp.1.vars(IP,c("G_Zonulin_0","L_Zonulin_0","P_Zonulin_0"),"Basal Zonulin")
p2<-bp.1.vars(IP,c("AUC_Zonulin_Glu","AUC_Zonulin_Lipids","AUC_Zonulin_Prots"),"Postpradial Zonulin")
grid.arrange(p1,p2)
```


Basal levels of `zonulin` are higher than Postpradial levels. The mean basal levels are higher in obese subjects, but higher in obese women with PCOS. This values are higher at basal levels on the glucose load day. Similar values at Postpradial time or non-obese subjects is observed. Meanwhile on obese subjects these values increases. At glucose load day, the higher levels are in obese men. Next day and, in overall, obese control females have the highest concentration of `zonulin`, meanwhile the last day again, obese men have the higher value.

##### Succinate.

```{r succinate}
lbp_basal<-bp.interaction[29][[1]]
p1<-bp.1.vars(IP,c("G_Succinate_0","L_Succinate_0","P_Succinate_0"),"Basal Succinate" )
p2<-bp.1.vars(IP,c("AUC_Succ_Glu","AUC_Succ_Lipids","AUC_Succ_Prots"),"Postpradial Succinate")+ theme(axis.text.x = element_text(angle = 0, vjust = 0.9, hjust=1,size = 7))
grid.arrange(p1,p2)
```
Basal concentration levels of `succinate` are higher than Postprandial levels. This levels are higher in obese subjects, particularly on PCOS women. At Postpradial time, non-obese males, have higher values, being this concentration higher on the first day. On obese subjects, also, the higher values are at the first day, with PCOS obese females with higher concentration, even in comparison with non-obese subjects. 

```{r}
//
```

## Statistical multivariate descriptive analysis.

We are going to perform a dimensionality reduction approach. Because, we are only interested on the visualization of the variables, we did not perform corroboration of the assumptions underlying PCA. Apart from the IP markers, we are also going to compute the PCA with the BMi variable.




```{r}
Y<-IP[,6:ncol(IP)]
pcx<-prcomp(Y,scale=T,center=T)

   X.scores <- pcx$x
    X.loadings <- pcx$rot
    colnames(X.scores)<-paste0("PC",1:dim(X.scores)[2])
    
    colnames(X.loadings) <- paste0("PC",1:dim(X.loadings)[2])
    varianza<-as.data.frame( 100*pcx$sdev^2/sum(pcx$sdev^2))
    varianza$PC <- paste0("PC",1:length(100*pcx$sdev^2/sum(pcx$sdev^2)))
    colnames(varianza)<-c("varianza","PC")
  # scores
  X.scores <- as.data.frame(X.scores)
  Group <- IP$GROUP
  Group2<-IP$OBESE
  colores<-1:length(levels(Group))
   # main plot
   p1 <- ggplot(X.scores,aes(x=PC1, y=PC2)) +
     theme_article() +
     geom_hline(yintercept = 0, color = "gray70") +
     geom_vline(xintercept = 0, color = "gray70") +
     geom_point(aes(color = Group,shape=Group2), alpha = 0.55, size = 3) +
     coord_cartesian(xlim = c(min(X.scores[,1]),max(X.scores[,1]))) +
     scale_fill_discrete(name = "Group")+
     labs(x = c(paste("PC1",round(varianza[1,1],2),"%")),y=c(paste("PC2",round(varianza[2,1],2),"%"))) +  
     ggtitle(paste("Scores BMI included",sep=" "))+ 
     theme(plot.title = element_text(hjust = 0.5)) +
     scale_color_manual(values=colores)+ theme(legend.position = "none")

   X.loadings <- as.data.frame(X.loadings)
   # main plot
   p2.1 <- ggplot(X.loadings,aes(x=PC1, y=PC2,label = colnames(Y))) +
     theme_classic() +
     geom_hline(yintercept = 0, color = "gray70") +
     geom_vline(xintercept = 0, color = "gray70") +
     geom_point(aes(color = "gray70"), alpha = 0.55, size = 3) +
     coord_cartesian(xlim = c(min(X.loadings[,1]),max(X.loadings[,1]))) +
     geom_text_repel(aes(label = colnames(Y)),segment.size = 0.2, size = 3) + 
     labs(x = c(paste("PC1",round(varianza[1,1],2),"%")),y=c(paste("PC2",round(varianza[2,1],2),"%"))) +  
     ggtitle(paste("","Loadings BMI included",sep=" "))+ 
     theme(plot.title = element_text(hjust = 0.5))+ theme(legend.position = "none")

X<-IP[,7:ncol(IP)]
  
 pcx<-prcomp(X,scale=T,center=T)
     


    X.scores <- pcx$x
    X.loadings <- pcx$rot
    colnames(X.scores)<-paste0("PC",1:dim(X.scores)[2])
    
    colnames(X.loadings) <- paste0("PC",1:dim(X.loadings)[2])
    varianza<-as.data.frame( 100*pcx$sdev^2/sum(pcx$sdev^2))
    varianza$PC <- paste0("PC",1:length(100*pcx$sdev^2/sum(pcx$sdev^2)))
    colnames(varianza)<-c("varianza","PC")
  # scores
  X.scores <- as.data.frame(X.scores)
  Group <- IP$GROUP
  Group2<-IP$OBESE
  colores<-1:length(levels(Group))
   # main plot
   p1_1 <- ggplot(X.scores,aes(x=PC1, y=PC2)) +
     theme_article() +
     geom_hline(yintercept = 0, color = "gray70") +
     geom_vline(xintercept = 0, color = "gray70") +
     geom_point(aes(color = Group,shape=Group2), alpha = 0.55, size = 3) +
     coord_cartesian(xlim = c(min(X.scores[,1]),max(X.scores[,1]))) +
     scale_fill_discrete(name = "Group")+
     labs(x = c(paste("PC1",round(varianza[1,1],2),"%")),y=c(paste("PC2",round(varianza[2,1],2),"%"))) +  
     ggtitle(paste("Scores BMI not included",sep=" "))+ 
     theme(plot.title = element_text(hjust = 0.5)) +
     scale_color_manual(values=colores)

   X.loadings <- as.data.frame(X.loadings)
   # main plot
   p2_2 <- ggplot(X.loadings,aes(x=PC1, y=PC2,label = colnames(X))) +
     theme_classic() +
     geom_hline(yintercept = 0, color = "gray70") +
     geom_vline(xintercept = 0, color = "gray70") +
     geom_point(aes(color = "gray70"), alpha = 0.55, size = 3) +
     coord_cartesian(xlim = c(min(X.loadings[,1]),max(X.loadings[,1]))) +
     geom_text_repel(aes(label = colnames(X)),segment.size = 0.1, size = 2) + 
     labs(x = c(paste("PC1",round(varianza[1,1],2),"%")),y=c(paste("PC2",round(varianza[2,1],2),"%"))) +  
     ggtitle(paste("","Loadings BMI not included",sep=" "))+ 
     theme(plot.title = element_text(hjust = 0.5))+ theme(legend.position = "none")
    

  
  
    graficos <- ggarrange(p1,p2.1,common.legend = T,legend = "bottom")
  

  
  q1<-annotate_figure(graficos, top = text_grob("Classic PCA IP data", 
               color = "black", face = "bold", size = 14))
  q1


```


```{r}
X<-IP[,6:ncol(IP)]
res<-plotPCA_scores(X,list(IP$OBESE,IP$GROUP),title="IP",center_ = T,scale_ = T,labels=IP$Paciente)
res[[2]]
```

```{r}
  fviz_pca_var(res[[1]],
    col.var = "contrib", # Color by contributions to the PC
    gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07"),

  repel = TRUE
# Avoid text overlapping
)
```





